src/components/Popover.tsx

```diff
@@ -1,15 +1,41 @@
-import React from "react";
+import React, { useLayoutEffect, useRef } from "react";
 
 type Props = {
   top?: number;
   left?: number;
   children?: React.ReactNode;
   onCloseRequest?(): void;
+  fitInViewport?: boolean;
 };
 
-export function Popover({ children, left, onCloseRequest, top }: Props) {
+export function Popover({
+  children,
+  left,
+  top,
+  onCloseRequest,
+  fitInViewport = false
+}: Props) {
+  const popoverRef = useRef(null);
```

Comment:
You can specify parameter type of useRef explicitly `useRef<HTMLDivElement>(null)`. So there would be no need in type casts later

Reply:
If I do that though, I get a TS error when passing the ref to the div as prop:

```
Type 'MutableRefObject<HTMLElement | null>' is not assignable to type 'string | ((instance:
HTMLDivElement | null) => void) | RefObject<HTMLDivElement> | null | undefined'.
  Type 'MutableRefObject<HTMLElement | null>' is not assignable to type 'RefObject<HTMLDivElement>'.
    Types of property 'current' are incompatible.
      Type 'HTMLElement | null' is not assignable to type 'HTMLDivElement | null'.
        Property 'align' is missing in type 'HTMLElement' but required in type 'HTMLDivElement'.  TS2322
```

Reply:
you have to pass exactly HTMLDivElement (not the HTMLElement)

Reply:
Force pushed with the fix. Thanks.